File: cmd_daily_task.py
import random
import db_manager

COMMANDS = ["مهمتي"]

# ======================
# أمر: مهمتي
# ======================
def handle(bot, message):
    uid = message.from_user.id
    text = message.text.strip()

    if text != "مهمتي":
        return

    task = db_manager.get_daily_task(uid)

    # ===== لا توجد مهمة + لم يحن الوقت =====
    if task is None:
        remaining = db_manager.time_left_for_task(uid)
        if remaining:
            bot.reply_to(
                message,
                "⏳ لا يمكنك أخذ مهمة الآن\n\n"
                f"🕒 الوقت المتبقي:\n{remaining}\n\n"
                "اصبر… صندوق الحظ يحب الصابرين 🎁"
            )
        else:
            bot.reply_to(message, "❌ لا توجد مهمة حالياً.")
        return

    # ===== عرض المهمة الحالية =====
    bot.reply_to(
        message,
        f"🎯 مهمتك لليوم:\n"
        f"{task['desc']}\n\n"
        "✨ بعد إتمامها ستحصل على صندوق الحظ النادر 🎁"
    )


# ======================
# تُستدعى من الألعاب
# ======================
def check_task_completion(bot, message, mission_type):
    """
    mission_type:
    - dice
    - roulette
    - أي لعبة مستقبلية
    """

    uid = message.from_user.id

    completed = db_manager.complete_mission(uid, mission_type)

    if not completed:
        return

    db_manager.add_to_inventory(uid, "🎁 صندوق الحظ النادر")

    bot.send_message(
        message.chat.id,
        "✅ تم إكمال مهمتك اليومية بنجاح!\n\n"
        "🎁 تم إرسال صندوق الحظ النادر إلى مخزونك\n"
        "📦 اكتب: مخزوني لعرضه"
    )

File: cmd_daily_task.js
// ملف: cmd_daily_task.js

const COMMANDS = ["مهمتي", "انجزت"];

function handle(bot, ctx, db) {
    if (!ctx.message || !ctx.message.text) return;

    const uid = ctx.from.id;
    const text = ctx.message.text.trim();

    // ======================
    // عرض المهمة
    // ======================
    if (text === "مهمتي") {
        const task = db.get_daily_task(uid);

        // لا توجد مهمة + وقت انتظار
        if (!task) {
            const remaining = db.time_left_for_task(uid);
            if (remaining) {
                ctx.reply(
                    "⏳ لا يمكنك أخذ مهمة الآن\n\n" +
                    `🕒 الوقت المتبقي:\n${remaining}\n\n` +
                    "اصبر… صندوق الحظ يحب الصابرين 🎁"
                );
            } else {
                ctx.reply("❌ لا توجد مهمة حالياً.");
            }
            return;
        }

        // عرض المهمة
        ctx.reply(
            "🎯 مهمتك لليوم:\n" +
            `${task.desc}\n\n` +
            "✍️ بعد إنجازها اكتب:\n" +
            "👉 انجزت"
        );
        return;
    }

    // ======================
    // إنجاز المهمة يدويًا
    // ======================
    if (text === "انجزت") {
        const task = db.get_daily_task(uid);

        if (!task) {
            ctx.reply("❌ ما عندك مهمة اليوم.");
            return;
        }

        const completed = db.complete_mission(uid, "manual");

        if (!completed) {
            ctx.reply("⚠️ مهمتك منجزة مسبقًا أو غير صالحة.");
            return;
        }

        db.add_to_inventory(uid, "🎁 صندوق الحظ النادر");

        ctx.reply(
            "✅ تم إنجاز مهمتك اليومية بنجاح!\n\n" +
            "🎁 حصلت على صندوق الحظ النادر\n" +
            "📦 اكتب: مخزوني لعرضه"
        );
        return;
    }
}

// ======================
// تُستدعى من الألعاب
// ======================
function check_task_completion(bot, ctx, db, mission_type) {
    const uid = ctx.from.id;

    const completed = db.complete_mission(uid, mission_type);
    if (!completed) return;

    db.add_to_inventory(uid, "🎁 صندوق الحظ النادر");

    ctx.reply(
        "✅ تم إكمال مهمتك اليومية!\n\n" +
        "🎁 صندوق الحظ النادر أُضيف لمخزونك\n" +
        "📦 اكتب: مخزوني"
    );
}

module.exports = {
    handle,
    check_task_completion
};
